src/pages/app/index.tsx

```diff
@@ -1,81 +1,21 @@
-import InputBox from '@/components/InputBox';
+import InputSection from '@/components/App/InputSection';
+import OutputSection from '@/components/App/OutputSection';
 import Layout from '@/components/Layout';
 import Toast from '@/components/Toast';
+import { urlRegex } from '@/constants/constants';
 import { TINY_SITE } from '@/constants/url';
 import IsAuthenticated from '@/hooks/isAuthenticated';
-import { urlRegex } from '@/utils/constants';
 import shortenUrl from '@/utils/shortenUrl';
 
-import CopyIcon from '../../components/icons/copy';
-import ShareIcon from '../../components/icons/share';
-
-interface InputSectionProps {
-    url: string;
-    setUrl: (url: string) => void;
-    handleUrl: () => void;
-}
-
-interface OutputSectionProps {
-    shortUrl: string;
-    handleCopyUrl: () => void;
-}
-
-const InputSection: React.FC<InputSectionProps> = ({ url, setUrl, handleUrl }) => (
-    <div className="bg-gray-200 flex flex-row justify-center items-center space-y-0 space-x-0 rounded-2xl mt-5 sm:mt-10">
-        <InputBox
-            type="text"
-            hideLabel={true}
-            className="bg-gray-200 w-full outline-none p-4 rounded-l-2xl"
-            onChange={(e: ChangeEvent<HTMLInputElement>) => setUrl(e.target.value)}
-            value={url}
-            placeholder="🔗 Enter the URL"
-            name="URL"
-        />
-        <Button className="bg-gray-300 rounded-r-2xl p-4 hover:bg-gray-400" onClick={handleUrl}>
-            Generate
-        </Button>
-    </div>
-);
-
-const OutputSection: React.FC<OutputSectionProps> = ({ shortUrl, handleCopyUrl }) => (
-    <div className="bg-gray-200 flex flex-row justify-center items-center space-y-0 space-x-0 rounded-2xl mt-2">
-        <InputBox
-            type="text"
-            name="URL"
-            hideLabel={true}
-            className="bg-gray-200 w-full outline-none p-4 rounded-l-2xl"
-            value={shortUrl}
-            placeholder="Copy the URL"
-        />
-        <a
-            type="button"
-            className="bg-gray-200  px-2 py-4 hover:bg-gray-400"
-            href={shortUrl}
-            target="_blank"
-            rel="noopener noreferrer"
-        >
-            <ShareIcon />
-        </a>
-        <Button
-            type="button"
-            className="bg-gray-200 rounded-r-2xl px-2 py-4 hover:bg-gray-400"
-            testId="copy-button"
-            onClick={handleCopyUrl}
-        >
-            <CopyIcon />
-        </Button>
-    </div>
-);
-
 const App = () => {
     const [url, setUrl] = useState<string>('');
     const [shortUrl, setShortUrl] = useState<string>('');
     const [toastMessage, setToastMessage] = useState<string>('');
     const [showToast, setShowToast] = useState<boolean>(false);
-    const [showInputBox, setShowInputBox] = useState<boolean>(false);
+    const [showInputBox, setShowInputBox] = useState<boolean>(true);
+    const [showOutputBox, setShowOutputBox] = useState<boolean>(false);
```

Comment:
here also

Reply:
Using the "Add a Suggestion" GitHub feature to suggest to me how to reduce use useState please

Reply:
const [state, setState] = useState({
        url: '',
        shortUrl: '',
        toastMessage: '',
        showToast: false,
        showInputBox: true,
        showOutputBox: false,
    });
    
    something like this

Reply:
I believe using multiple useState hooks is more readable than what you suggested.

Reply:
It is perfectly fine to use multiple useState hooks as needed.